geo/geomfn: implement ST_Centroid for Geometry and string

otan opened the pull request:
Also had to add hacks in to handle `POINT EMPTY` decoding -- this is
currently a problem with both GEOS and twpayne/go-geom, which I have
circumvented for now. An issue has been filed to track this.

Resolves https://github.com/cockroachdb/cockroach/issues/48893.
Resolves https://github.com/cockroachdb/cockroach/issues/48805.

Release note (sql change): Implement the ST_Centroid builtin which works
for Geometry and string arguments.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49613)
<!-- Reviewable:end -->

otan committed:
geo/geomfn: implement ST_Centroid for Geometry and string

Also had to add hacks in to handle `POINT EMPTY` decoding -- this is
currently a problem with both GEOS and twpayne/go-geom, which I have
circumvented for now. An issue has been filed to track this.

Release note (sql change): Implement the ST_Centroid builtin which works
for Geometry and string arguments.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 9 of 9 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49613#-:-M8NlsMj72MhWHj4_Jtp:b593w57)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
TFTR!

bors r=sumeerbhola
